### PR TITLE
Add profile page for user

### LINKED ISSUE #27: Add additional account information
- [x] username
- [x] firstname, lastname
- [x] place of residence
- [x] birth date

### LINKED ISSUE #28: Add option to set current status
- [x] available
- [x] working

### LINKED ISSUE #30: Add profile page for account
- [ ] contains information described #27 
- [ ] should be linked in contact list #26

### PR DESCRIPTION
**Pull request by frcroth**

### CODE REVIEW

**Comment by frcroth** [comment]
Fixes #30

**Comment by frcroth** [comment]
This pull request includes #27 #28 #30

File: app/models/user.rb [hunk 1]
```diff
@@ -8,4 +8,21 @@ class User < ApplicationRecord
...
+
+  validates :username, :email, presence: true
+  validates :current_status, inclusion: { in: VALID_STATUS_LIST }
+
+  after_initialize :init
+
+  attribute :current_status, :string, default: "available"
+
+  def init
+    self.username ||= email.split('@', 2)[0]
```
Comment (mpass99) [inline app/models/user.rb:26]: Does this mean that admin@example.com and admin@test.de get the same username?
Is it ok that multiple users have the same username?

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -9,13 +9,39 @@
...
   end
+
+  it "is not valid without a username" do
+    @user.username = ""
+    expect(@user).not_to be_valid
+  end
+
+  it "creates user name from email" do
+    user = described_class.new({ email: "test-user@example.org" })
+    expect(user.username).to eq("test-user")
```
Comment (mpass99) [inline spec/models/user_spec.rb:40]: See above

File: app/models/user.rb (hunk 1 above)
Comment (frcroth) [reply app/models/user.rb:26]: Yes.
It was not specified that usernames should be unique.

File: app/views/users/_status_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<%= form_for(@user, local: true, url: update_status_user_path(@user)) do |f| %>
+  <div class="field">
+    <%= f.label :current_status %>
+    <%= f.select(:current_status, @user.select_status_list) %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Change current status" %>
```
Comment (sirkrypt0) [inline app/views/users/_status_form.html.erb]: Internationalization is missing here.

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  <tr>
+    <td>Birthdate:</td>
+    <td><%= @user.birthdate %></td>
+  </tr>
+</table>
+
+<% if user_signed_in? && current_user.id == @user.id %>
+  <%= render 'status_form' %>
+<% else %>
+  <p>Current status: <%= @user.current_status %></p>
```
Comment (sirkrypt0) [inline app/views/users/show.html.erb]: Internationalization is missing here.

File: spec/features/user_registration_spec.rb [hunk 5]
```diff
@@ -21,4 +23,16 @@
     # See config/locales/devise.en.yml
     expect(page).to have_content(I18n.t 'devise.registrations.signed_up')
   end
+
+  it "can not signup without matching passwords" do
+    within '#new_user' do
+      user_attrs = FactoryBot.attributes_for(:user)
+      fill_in "Email", with: user_attrs[:email]
+      fill_in "Password", with: user_attrs[:password]
+      fill_in "Password confirmation", with: "wrong_password"
```
Comment (sirkrypt0) [inline spec/features/user_registration_spec.rb:32]: Internationalization

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]
lgtm, only internationalization is missing in some places.

**Comment by frcroth** [comment]
This pull request now also contains #34 (already approved)

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  <tr>
+    <td><%= I18n.t('user.attributes.birthdate') %></td>
+    <td><%= @user.birthdate %></td>
+  </tr>
+</table>
+
+<% if user_signed_in? && current_user.id == @user.id %>
+  <%= render 'status_form' %>
+<% else %>
+  <p>Current status: <%= @user.current_status %></p>
```
Comment (felixauringer) [inline app/views/users/show.html.erb]: We should use `I18n` here.

File: spec/requests/application_spec.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe "Application", type: :request do
+
+  it 'renders a 404 if active record model is not found' do
+    get '/users/-1'
```
Comment (felixauringer) [inline spec/requests/application_spec.rb]: maybe we should use the `user_path(-1)` helper here (if that's possible)

**Review by felixauringer — COMMENTED** [review]
We do not have tests for the `edit` view. In my opinion it does not have to be fixed in this PR as it would be nice to merge this as soon as possible.

**Pull request merged**